### PR TITLE
[#4520] Imporvement(jdbc-catalogs): Shrink JDBC catalog binary package size to about 3MB.

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Remove some unnecessary jar dependencies in the runtime classpath for JDBC catalogs.

### Why are the changes needed?

Shrink JDBC catalog package.

Fix: #4520 

### Does this PR introduce _any_ user-facing change?

N/A.
### How was this patch tested?

Test locally and CI.

MySQL catalog:
image

PostgreSQL catalog:

image
Doris catalog:

image

### CODE REVIEW

File: catalogs/catalog-jdbc-mysql/build.gradle.kts [hunk 1]
```diff
@@ -25,30 +25,36 @@ plugins {
...
+    exclude(group = "*")
+  }
+  implementation(project(":common")) {
+    exclude(group = "*")
+  }
+  implementation(project(":core")) {
+    exclude(group = "*")
+  }
 
   implementation(libs.bundles.log4j)
```
Comment (mchades) [inline catalogs/catalog-jdbc-mysql/build.gradle.kts:44]: Can remove this and keep `implementation(libs.slf4j.api)` ?

Comment (yuqi1129) [reply catalogs/catalog-jdbc-mysql/build.gradle.kts:44]: No, there will be compilation errors if we remove the dependency.

Comment (mchades) [reply catalogs/catalog-jdbc-mysql/build.gradle.kts:44]: alright

File: catalogs/catalog-jdbc-doris/build.gradle.kts [hunk 2]
```diff
@@ -61,7 +65,11 @@ tasks {
   }
   val copyCatalogLibs by registering(Copy::class) {
     dependsOn("jar", "runtimeJars")
-    from("build/libs")
+    from("build/libs") {
+      exclude("guava-*.jar")
+      exclude("log4j-*.jar")
+      exclude("slf4j-*.jar")
+    }
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-doris/build.gradle.kts:72]: Why do we need to remove these jars explicitly? If we're using them only in compile, maybe we can change from "implementation" to "compileOnly"?

Comment (yuqi1129) [reply catalogs/catalog-jdbc-doris/build.gradle.kts:72]: Those jars already exist in the server class path, duplicated jars are useless.

Comment (yuqi1129) [reply catalogs/catalog-jdbc-doris/build.gradle.kts:72]: >  If we're using them only in compile, maybe we can change from "implementation" to "compileOnly"?

Yeah, let me have a try.

Comment (yuqi1129) [reply catalogs/catalog-jdbc-doris/build.gradle.kts:72]: > > If we're using them only in compile, maybe we can change from "implementation" to "compileOnly"?
> 
> Yeah, let me have a try.

It seems that making it `compileOnly` can't pass tests and runs into classes not found problem.

File: catalogs/catalog-jdbc-mysql/build.gradle.kts [hunk 3]
```diff
@@ -25,30 +25,36 @@ plugins {
...
-  implementation(project(":catalogs:catalog-common"))
-  implementation(project(":catalogs:catalog-jdbc-common"))
-  implementation(project(":common"))
-  implementation(project(":core"))
+  implementation(project(":api")) {
+    exclude(group = "*")
+  }
+  implementation(project(":catalogs:catalog-common")) {
+    exclude(group = "*")
+  }
```
Comment (jerryshao) [inline catalogs/catalog-jdbc-mysql/build.gradle.kts:33]: Why only mysql relies on "catalog-common"?

Comment (yuqi1129) [reply catalogs/catalog-jdbc-mysql/build.gradle.kts:33]: Only MySQL use `MysqlConstants.GRAVITINO_AUTO_INCREMENT_OFFSET_KEY` and support the feature `auto_increment` now.

**Review by jerryshao — APPROVED** [review]

**Pull request merged**